modules/azure/sql.go

```diff
@@ -0,0 +1,185 @@
+	require.NoError(t, err)
+
+	return serverState
+}
+
+// GetSQLServerStateE is a helper function that gets the sql server state
+func GetSQLServerStateE(t testing.TestingT, subscriptionID string, resGroupName string, serverName string) (sql.ServerState, error) {
+	// Create a SQl Server client
+	sqlClient, err := GetSQLServerClient(subscriptionID)
+	if err != nil {
+		return "", err
+	}
+
+	// Get the corresponding server client
+	sqlServer, err := sqlClient.Get(context.Background(), resGroupName, serverName)
+	if err != nil {
+		return "", err
+	}
+
+	//Return server state
+	return sqlServer.State, nil
+}
+
+// GetDatabaseClient  is a helper function that will setup a sql DB client
+func GetDatabaseClient(subscriptionID string) (*sql.DatabasesClient, error) {
+	// Validate Azure subscription ID
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a sql DB client
+	sqlDBClient := sql.NewDatabasesClient(subscriptionID)
+
+	// Create an authorizer
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	// Attach authorizer to the client
+	sqlDBClient.Authorizer = *authorizer
+
+	return &sqlDBClient, nil
+}
+
+// GetDatabaseID is a helper function that gets the sql db id
+func GetDatabaseID(t testing.TestingT, resGroupName string, serverName string, dbName string, subscriptionID string) string {
+	dbID, err := GetDatabaseIDE(t, subscriptionID, resGroupName, serverName, dbName)
+	require.NoError(t, err)
+
+	return dbID
+}
+
+// GetDatabaseIDE is a helper function that gets the sql db id
+func GetDatabaseIDE(t testing.TestingT, subscriptionID string, resGroupName string, serverName string, dbName string) (string, error) {
+	// Create a SQl db client
+	sqlClient, err := GetDatabaseClient(subscriptionID)
+	if err != nil {
+		return "", err
+	}
+
+	//Get the corresponding DB client
+	sqlDb, err := sqlClient.Get(context.Background(), resGroupName, serverName, dbName, "")
+	if err != nil {
+		return "", err
+	}
+
+	// Return DB ID
+	return *sqlDb.ID, nil
+}
+
+// GetDatabaseStatus is a helper function that gets the sql db state
+func GetDatabaseStatus(t testing.TestingT, resGroupName string, serverName string, dbName string, subscriptionID string) string {
+	dbStatus, err := GetDatabaseStatusE(t, subscriptionID, resGroupName, serverName, dbName)
+	require.NoError(t, err)
+
+	return dbStatus
+}
+
+// GetDatabaseStatusE is a helper function that gets the sql db state
+func GetDatabaseStatusE(t testing.TestingT, subscriptionID string, resGroupName string, serverName string, dbName string) (string, error) {
+	// Create a SQl db client
+	sqlClient, err := GetDatabaseClient(subscriptionID)
+	if err != nil {
+		return "", err
+	}
+
+	//Get corresponding DB client
+	sqlDb, err := sqlClient.Get(context.Background(), resGroupName, serverName, dbName, "")
+	if err != nil {
+		return "", err
+	}
+
+	//Return DB status
+	return *sqlDb.Status, nil
+}
```

Comment:
Add a method to get a list of databases on the sql server.

Reply:
added ListSQLServerDatabasesE